source/multigrid/mg_tools.cc

```diff
@@ -1547,6 +1547,32 @@ namespace MGTools
       }
 
   }
+
+
+
+  template <int dim, int spacedim>
+  unsigned int
+  max_level_for_coarse_mesh (const Triangulation<dim,spacedim> &tria)
+  {
+    // Find minimum level for an active cell in
+    // this locally owned subdomain
+    int min_level = tria.n_global_levels();
+    typename Triangulation<dim,spacedim>::active_cell_iterator cell = tria.begin_active(),
+                                                               endc = tria.end();
+    for (; cell!=endc; ++cell)
+      if (cell->is_locally_owned())
+        if (cell->level() < min_level)
+          min_level = cell->level();
+
+    // Communicate to find minimum level for an
+    // active cell over all subdomains
+    unsigned int global_min = Utilities::MPI::min(min_level, MPI_COMM_WORLD);
```

Comment:
Don’t hardcode MPI_COMM_WORLD. Take communicator from p::Tria, don’t do anything for standard tria.